### PR TITLE
Pass more parameters to player's constructor

### PR DESCRIPTION
**Pull request by ceyusa**
This pull request is mainly about passing more parameters to player's constructor.

Also it includes a patch to make PlayerGLContext's enums as seriable in order to share them through IPC.

Finally, a cosmetic change which removes currently unused external crates.

### CODE REVIEW

**Comment by ceyusa** [comment]
@ferjm r?

File: player/lib.rs [hunk 1]
```diff
@@ -69,7 +69,9 @@ pub enum StreamType {
 }
 
 pub trait Player: Send {
+    /// Register an IpcSender that will push PlayerEvents generated by the playerq
```
Comment (ferjm) [inline player/lib.rs]: typo: `playerq`

**Comment by ceyusa** [comment]
> In order to avoid future delays, ask for review again after you force push, please.

Ok!  Thanks :)

**Pull request merged**